Readability.js

```diff
@@ -1655,8 +1655,6 @@ Readability.prototype = {
           toRemove = true;
         } else if (!isList && weight < 25 && linkDensity > 0.2) {
           toRemove = true;
-        } else if (weight >= 25 && linkDensity > 0.5) {
-          toRemove = true;
```

Comment:
This change affects more than just list items, as evidenced by the buzzfeed testcase. I think we need to tread slightly more carefully here.

I think an alternative fix that might well work pretty well is not counting links that are anchor links to the same page (so linkDensity will be lower). As a bonus, we should try to keep hash links non-absolute so they work in reader mode.